db/builder.cc

```diff
@@ -217,6 +213,13 @@ Status BuildTable(
     if (io_status->ok() && !empty) {
       *io_status = file_writer->Close();
     }
+    if (io_status->ok() && !empty) {
+      // Add the checksum information to file metadata.
+      meta->file_checksum = builder->GetFileChecksum();
+      meta->file_checksum_func_name = builder->GetFileChecksumFuncName();
+    }
+    delete builder;
```

Comment:
I don't think we have to move `delete builder` to here. The checksum and checksum name can be directly read from `file_writer`. I didn't see anything wrong with moving builder here, but if we don't have to expose ourselves with the risk, maybe we should consider so.

Reply:
@siying If so, the GetFilechecksum and GetFileChecksumFuncName interfaces in all table_bulider can be removed. Do you think we should keep them in table_builder?

Reply:
+1 for moving the `delete builder` back to line 210 before calling Sync and Close.